internal/cmd/gocdk/provision.go

```diff
@@ -54,12 +60,170 @@ func registerProvisionCmd(ctx context.Context, pctx *processContext, rootCmd *co
 	rootCmd.AddCommand(provisionCmd)
 }
 
+const provisionStaticRootDir = "/provision"
+
 // The "provision list" command.
-func provisionList(pctx *processContext) {
-	pctx.Logf("NOT YET IMPLEMENTED")
+func provisionList(pctx *processContext) error {
+	// TODO(rvangent): There must be an easier way to get a listing... add
+	// a helper to static.
+	dir, err := static.Open(provisionStaticRootDir)
+	if err != nil {
+		return xerrors.Errorf("provision list: couldn't open static resources: %w", err)
+	}
+	ptInfos, err := dir.Readdir(-1)
+	if err != nil {
+		return xerrors.Errorf("provision list: couldn't read static resources: %w", err)
+	}
+	// The entries at this level are portable APIs, e.g., "blob".
+	for _, ptInfo := range ptInfos {
+		ptName := ptInfo.Name()
+		if !ptInfo.IsDir() {
+			continue
+		}
+		ptTypeDir, err := static.Open(path.Join(provisionStaticRootDir, ptName))
+		if err != nil {
+			return xerrors.Errorf("provision list: couldn't open static resources: %w", err)
+		}
+		providerInfos, err := ptTypeDir.Readdir(-1)
+		if err != nil {
+			return xerrors.Errorf("provision list: couldn't read static resources: %w", err)
+		}
+		for _, info := range providerInfos {
+			if !info.IsDir() {
+				continue
+			}
+			pctx.Println(path.Join(ptName, info.Name()))
+		}
+	}
+	return nil
 }
 
 // The "provision add" command.
+// TODO(rvangent): Can we support adding a particular type more than once?
+// TODO(rvangent): If things fail in the middle, we are in an undefined state.
+//                 Unclear how to handle that....
+// TODO(rvangent): Modifying Terraform files in place means that we need to run
+//                 "terraform init" again; currently we don't.
```

Comment:
Likely the user won't feel the effects of this most of the time because `gocdk apply` runs `terraform init`. The places where it doesn't right now are a bug: #2291.

Reply:
Updated to reference the bug.